x509-identity-mgmt/js/src/routes/certificates-router.js

```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+
+const HttpStatus = require('http-status-codes');
+
+const paginate = require('express-paginate');
+
+const { validateRegOrGenCert, validateChangeOwnerCert } = require('../core/schema-validator');
+
+const service = require('../services/certificates-service');
+
+const { certificate: parser } = require('../db');
+
+const router = express.Router();
+
+router.route('/certificates')
+  /* Generate x.509 Certificate from CSR
+   * (or also)
+   * Register x.509 Certificate Issued by an External CA */
+  .post(validateRegOrGenCert(), async (req, res) => {
+    let result = null;
+    if (!req.body.belongsTo) {
```

Comment:
I think we need a stronger validation for the belongsTo. In case, a certificate is assigned to a device, we need to guarantee that this device exists in the dojot device management. Furthermore, the device management cannot allow to remove a device if a certificate is associated with it. The problem is that the information is distributed among multiple services.

Reply:
I thought about it, it's even in the API architecture document, but I haven't implemented it because we are not working with this attribute yet. When we evolve this component, there will certainly be this verification, but for this version, we were able to work without.